### PR TITLE
Core/Misc: Update locales_achievement_reward to simple system

### PR DESCRIPTION
**Pull request by ChazyTheBest**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

- Update locales_achievement_reward to simple system
- Update achievement_reward to keep consistency

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** None

**Tests performed:** (Does it build, tested in-game, etc.)

- [x] Builds

### CODE REVIEW

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 1]
```diff
@@ -591,8 +591,8 @@ void AchievementMgr::LoadFromDB(PreparedQueryResult achievementResult, PreparedQ
 
             // title achievement rewards are retroactive
             if (AchievementReward const* reward = sAchievementMgr->GetAchievementReward(achievement))
-                if (uint32 titleId = reward->titleId[Player::TeamForRace(GetPlayer()->getRace()) == ALLIANCE ? 0 : 1])
```
Comment (Shauren) [inline src/server/game/Achievements/AchievementMgr.cpp]: stooooooooop right here, `uint32 titleId` is a local variable scoped to this if only, it should not start with uppercase (this comment applies to ALL local variables changed in this PR)

**Review by Shauren — CHANGES_REQUESTED** [review]

**Comment by Shauren** [comment]
And keep the inserts, even if the table is empty

File: sql/updates/world/3.3.5/2017_xx_xx_xx_world.sql [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+-- achievement_reward_locale
 DROP TABLE IF EXISTS `achievement_reward_locale`;
 CREATE TABLE IF NOT EXISTS `achievement_reward_locale` (
   `ID` MEDIUMINT(8) UNSIGNED NOT NULL DEFAULT '0',
   `Locale` VARCHAR(4) NOT NULL,
   `Subject` TEXT,
   `Text` TEXT,
   PRIMARY KEY (`ID`, `Locale`)
-) ENGINE=MYISAM DEFAULT CHARSET=utf8;
+) ENGINE=MYISAM DEFAULT CHARSET=utf8mb4;
```
Comment (Shauren) [inline sql/updates/world/3.3.5/2017_xx_xx_xx_world.sql]: I am confused about this change, why?

File: src/server/game/Achievements/AchievementMgr.cpp (hunk 1 above)
Comment (ChazyTheBest) [inline src/server/game/Achievements/AchievementMgr.cpp]: right away

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 3]
```diff
@@ -1515,37 +1515,37 @@ void AchievementMgr::CompletedAchievement(AchievementEntry const* achievement)
...
-        MailDraft draft(reward->mailTemplate);
+        MailDraft draft(reward->MailTemplate);
 
-        if (!reward->mailTemplate)
+        if (!reward->MailTemplate)
         {
             // subject and text
-            std::string subject = reward->subject;
-            std::string text = reward->text;
+            std::string Subject = reward->Subject;
```
Comment (Shauren) [inline src/server/game/Achievements/AchievementMgr.cpp]: I said everywhere, this means also here

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 4]
```diff
@@ -1515,37 +1515,37 @@ void AchievementMgr::CompletedAchievement(AchievementEntry const* achievement)
...
+            std::string Text = reward->Text;
```
Comment (Shauren) [inline src/server/game/Achievements/AchievementMgr.cpp]: and here

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 5]
```diff
@@ -2686,38 +2682,38 @@ void AchievementGlobalMgr::LoadRewardLocales()
...
     }
 
     do
     {
         Field* fields = result->Fetch();
 
-        uint32 entry = fields[0].GetUInt32();
+        uint32 id               = fields[0].GetUInt32();
+        std::string localeName  = fields[1].GetString();
+        std::string Subject     = fields[2].GetString();
```
Comment (Shauren) [inline src/server/game/Achievements/AchievementMgr.cpp]: and here

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 6]
```diff
@@ -2686,38 +2682,38 @@ void AchievementGlobalMgr::LoadRewardLocales()
...
+        std::string Text        = fields[3].GetString();
```
Comment (Shauren) [inline src/server/game/Achievements/AchievementMgr.cpp]: and here

**Review by Shauren — CHANGES_REQUESTED** [review]

**Review by Shauren — APPROVED** [review]

**Pull request closed without merging**